### DISCUSSION TITLE
Flutter + Snapshot ?

### DISCUSSION THREAD

**Discussion by louis030195** [fastlane/fastlane#18602] [Q&A]
Hey, I am trying to find a way to automate the screenshot taking for app stores.
Fastlane Snapshot looks great but I don't want to write native code.

**Is there a way to use Snapshot with Flutter?**

PS: with such an automated process, how do you deal with authentication? Do you give a "hack authentication" to "screenshot takers code" I guess?

Thank you a lot in advance for your help :)

**Comment by Robert442** [comment]
Let's say you want to take a screenshot of the FlutterLogo widget . wrap it in a RepaintBoundary with will creates a separate display list for its child . and provide with a key

```
var scr= new GlobalKey();
RepaintBoundary(
         key: scr,
         child: new FlutterLogo(size: 50.0,))
```

and then you can get the pngBytes by converting the boundary to an image

```
takescrshot() async {
  RenderRepaintBoundary boundary = scr.currentContext.findRenderObject();
  var image = await boundary.toImage();
  var byteData = await image.toByteData(format: ImageByteFormat.png);
  var pngBytes = byteData.buffer.asUint8List();
  print(pngBytes);
  }
```


[MyAARPMedicare](https://www.myaarpmedicare.us)

**Comment by louis030195** [comment]
@Robert442 thanks! But I was more asking about Fastlane Snapshot integration, of course, I can do it all manually using https://pub.dev/packages/screenshot for example :)

**Reply by maxisme** [reply]
Yeah [screenshots](https://github.com/mmcc007/screenshots/issues/215) is the closest thing I have come to automating screenshots - but it [no longer works](https://github.com/mmcc007/screenshots/issues/215#issuecomment-826091549).

**Comment by Navil** [comment]
I am stuck at the same problem now. There is currently no good way to automate screenshots for a CI/CD pipeline. Using screenshots from Flutter integration tests does leave out the status bar on the top. So maybe it makes most sense having an option in frameit to add the status bar during the framing process.

**Comment by charlieforward9** [comment]
Is there an alternative to this these days? I was looking at fastlane but dont see it integrating with Flutter for screenshots well yet. Currently I have to run a session on each device type to generate the screenshots.

**Comment by FuadEfendi** [comment]
I am researching it too; maybe this will work (but I didn't try it yet): https://medium.com/@mregnauld/generate-screenshots-for-a-flutter-app-with-golden-testing-and-upload-them-to-the-stores-1-2-45f8df777aef

**Comment by FuadEfendi** [comment]
I am sticking to this, looks very logical & straightforward, no need to hack XCode and learn Ruby: https://docs.codemagic.io/knowledge-codemagic/flutter-screenshots-stores/
